### PR TITLE
Handle signals and user stdin more gracefully

### PR DESCRIPTION
**Pull request by Hoverbear**
Do a much better job handling signals, while not completing #41 , it progresses it.

### CODE REVIEW

**Comment by Hoverbear** [comment]
Failure due to #49.

File: src/cli/mod.rs [hunk 1]
```diff
@@ -40,3 +42,32 @@ impl CommandExecute for HarmonicCli {
         }
     }
 }
+
+pub(crate) async fn signal_channel() -> eyre::Result<(Sender<()>, Receiver<()>)> {
+    let (sender, reciever) = tokio::sync::broadcast::channel(100);
```
Comment (cole-h) [inline src/cli/mod.rs]: Does this `100` have any particular meaning, or was it chosen by the fair dice roll of xkcd fame?

File: src/cli/subcommand/install.rs [hunk 2]
```diff
@@ -4,10 +4,12 @@ use std::{
 };
 
 use crate::{
-    action::ActionState, cli::is_root, plan::RECEIPT_LOCATION, BuiltinPlanner, InstallPlan, Planner,
+    action::ActionState, cli::is_root, cli::signal_channel, plan::RECEIPT_LOCATION, BuiltinPlanner,
+    InstallPlan, Planner,
 };
 use clap::{ArgAction, Parser};
 use eyre::{eyre, WrapErr};
+use tokio_util::sync::CancellationToken;
```
Comment (cole-h) [inline src/cli/subcommand/install.rs:12]: This looks unused to me, but I could be wrong.

File: src/cli/mod.rs [hunk 3]
```diff
@@ -3,6 +3,8 @@ pub(crate) mod subcommand;
 
 use clap::Parser;
 use std::process::ExitCode;
+use tokio::sync::broadcast::{Receiver, Sender};
+use tokio_util::sync::CancellationToken;
```
Comment (cole-h) [inline src/cli/mod.rs:7]: Also here.

File: src/plan.rs [hunk 4]
```diff
@@ -1,6 +1,8 @@
 use std::path::PathBuf;
 
 use crossterm::style::Stylize;
+use tokio::sync::broadcast::Receiver;
+use tokio_util::sync::CancellationToken;
```
Comment (cole-h) [inline src/plan.rs:5]: And here.

**Review by cole-h — APPROVED** [review]
Oops, wrong button. Aside from my comments (nits) above, LGTM.

File: src/cli/mod.rs (hunk 1 above)
Comment (Hoverbear) [reply src/cli/mod.rs]: Fair dice roll!

File: src/cli/subcommand/install.rs (hunk 2 above)
Comment (Hoverbear) [reply src/cli/subcommand/install.rs:12]: It sure is!

**Pull request merged**